Makefile.am

```diff
@@ -116,7 +116,8 @@ setup.h: $(srcdir)/setup.h.in Makefile
 	     -e 's|@prefix[@]|$(prefix)|g' \
 	     -e 's|@p4includedir[@]|$(p4includedir)|g' \
 	     -e 's|@p4_14includedir[@]|$(p4_14includedir)|g' \
-	$< >$@
+	$< >$@.tmp
+	@if cmp $@ $@.tmp; then rm $@.tmp; else mv $@.tmp $@; fi
```

Comment:
OK. I'd recommend either moving it into a separate commit (so `git blame` explains the reasoning for the change) or adding a comment (so viewing the source explains the reasoning). Future readers of this code will struggle otherwise because Makefile interactions are nonlocal and are hard to reason about just from a diff.